src/jsonrpc2.ml

```diff
@@ -273,6 +273,20 @@ module Make (IO : IO) : S with module IO = IO = struct
           IO.return @@ J.from_string (Bytes.unsafe_to_string buf)
         in
         Log.debug (fun k -> k "got json %s" (J.to_string j));
+        (* Jsonrpc expect "params" to be object or array,
```

Comment:
That's really good. Would you mind extracting this to a toplevel function like, say, "repair_missing_params", and upgrade the comment to mention #20 ?